test/parse.test.ts

```diff
@@ -142,6 +142,16 @@ describe('parse', () =>{
         )
       })
     );
+    test(
+      `not (emails co "example.com" or emails co "example.org") and userType ne "Employee"`,
+      and(op("ne", "userType", "Employee"), {
+        op: "not",
+        filter: or(
+          op("co", "emails", "example.com"),
+          op("co", "emails", "example.org")
+        )
+      })
```

Comment:
This is the same AST as in the above test case but with the left and right side of the AND flipped in the SCIM string.